step/v2/step.go

```diff
@@ -243,3 +253,23 @@ func (p *pipeline) errExit(err error) {
 	p.failed = err
 	runtime.Goexit()
 }
+
+// cast performs a type cast from src to T.
+//
+// Unlike src.(T), this cast is valid when casting from an untyped nil to a typed nil.
+func cast[T any](src any) T {
```

Comment:
`cast` is a very overloaded word!

Let's give this function a more descriptive name, like `castWithTypedNil` or something like that. I know it's a library function, and I know we use it literally hundreds of times but we can be a bit more specific I think!

Reply:
I don't want to give the function a name like `castWithTypedNil`, since the point of the function is to hide the need to treat `nil`s differently from non-`nil` values. When we discover more problems with this function (and I'm sure we will), I don't want to change the function name. I could call it `assertType[T any](src any) T`, since go calls `src.(T)` a type assertion, but I'm not sure if that adds any clarity here.

Do you think that `assertType` is better?